migrate.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/sh
+
+set -e
+
+# Function to display usage
+usage() {
+    echo "Usage: $0 <operation> <network> <source_dir> [destination_dir]"
+    echo "  <operation>       Either pre or full"
+    echo "  <network>         Network name (celo-mainnet, alfajores, or baklava)"
+    echo "  <source_dir>      Source datadir directory (the value of the '--datadir' flag for the celo L1 client)"
+    echo "  [destination_dir] Optional destination datadir directory (should be used as the value for the '--datadir'"
+    echo "                    flag for the celo L2 client), if omitted './envs/<network>/datadir' will be used"
+    exit 1
+}
+
+# Check the number of arguments
+if [ "$#" -lt 3 ] || [ "$#" -gt 4 ]; then
+    usage
+fi
+
+# Assign positional arguments to variables
+operation=$1
+network=$2
+source_dir=$3
+destination_dir="${4:-./envs/$network/datadir}"
+
+# Validate the operation
+if [ "$operation" != "pre" ] && [ "$operation" != "full" ]; then
+    echo "Invalid operation: $operation"
+    usage
+fi
+
+# Validate network name
+if [ "$network" != "celo-mainnet" ] && [ "$network" != "alfajores" ] && [ "$network" != "baklava" ]; then
+    echo "Invalid network name: $network"
+    usage
+fi
+
+# Print parsed arguments
+echo "Network: $network"
+echo "Source Directory: $source_dir"
+echo "Destination Directory: $destination_dir"
+
+# Convert source and destination directories to absolute paths
+source_dir=$(readlink -f "$source_dir")
+destination_dir=$(readlink -f "$destination_dir")
+
+# Ensure destination directory exists for chaindata
+mkdir -p  "${destination_dir}/geth"
+
+if [ "${operation}" = "pre" ]; then
+  docker run --platform=linux/amd64 -it --rm \
+    -v "${source_dir}/celo/chaindata:/old-db" \
+    -v "${destination_dir}/geth/chaindata:/new-db" \
+    us-west1-docker.pkg.dev/devopsre/dev-images/cel2-migration-tool:5682b80ec60c47f582c6af8aa085ae6f9048d801 \
+    "${operation}" \
+      --old-db /old-db \
+      --new-db /new-db
+  exit 0
+fi
+
+# We need the OP_NODE__RPC_ENDPOINT to know where to connect to the L1 node.
+. "${network}.env"
```

Comment:
Since all vars are lowercase in this script and uppercase in the env file, this will not overwrite and variables and is safe.